Rename Greenplum to Greengage

KnightMurloc opened the pull request:
This patch replaces mentions of Greenplum with Greengage.
We are not touching ci, gppkg and copyright yet.
And also:
1. Mentions of Arenadata have been removed. Removed support for suffixes with
the Arenadata version.
2. Some dead links have been removed.
3. From the backup config 9-segment-db-replicated.tar.gz the arenadata suffix
has been removed.
4. The run_gpbackup_tests files.bash and Dockerfile have been moved from the
arenadata directory to ci.

whitehawk commented:
In the description:
'From the backup config 9-segment-db-replicated.tar.gz The arenadata suffix' -> 'From the backup config 9-segment-db-replicated.tar.gz the arenadata suffix'

whitehawk commented:
What are the plans for CI folder update?

KnightMurloc commented:
> What are the plans for CI folder update?

As I understand it, we don't use most of these files in our CI. Therefore, we need to review which ones we need and delete the others.

hilltracer commented on plugins/README.md:
```diff
@@ -32,7 +32,7 @@ options:
 
 ## Available plugins
-[gpbackup_s3_plugin](https://github.com/greenplum-db/gpbackup-s3-plugin): Allows users to back up their Greenplum Database to Amazon S3.
+[gpbackup_s3_plugin](https://github.com/GreengageDB/gpbackup-s3-plugin): Allows users to back up their Greengage Database to Amazon S3.
```
This link doesn't work

hilltracer reviewed (changes requested):
1. Should we change vmware mentions? e.g.:

https://github.com/GreengageDB/gpbackup/blob/45c612b6b077ffd92857c2a64018e945d9180322/CODE-OF-CONDUCT.md#L59

https://github.com/GreengageDB/gpbackup/blob/45c612b6b077ffd92857c2a64018e945d9180322/CONTRIBUTING.md#L12

https://github.com/GreengageDB/gpbackup/blob/516109fcb056bf5cf9c97e70c1bc217ee9ffc4f2/ci/regression/README.md#L13

Banch of words into ci/pivnet_release/metadata.yml gpbackup-release-generated.yml gpbackup-generated.yml


2. Should we save links like this:

https://github.com/GreengageDB/gpbackup/blob/e5fc71fdf7816d74a294b5d6c70fc3cd6567f14c/ci/regression/regression_pipeline.yml#L105

https://github.com/GreengageDB/gpbackup/blob/e5fc71fdf7816d74a294b5d6c70fc3cd6567f14c/ci/regression/regression_pipeline.yml#L112

3. What will we do with these test paths? Maybe it should be also changed like ${GOPATH}/src/github.com/GreengageDB/gpbackup?
<details>
<summary>spoiler</summary>

```sh
/home/den/gpdb/src/gpbackup/.github/workflows/build_and_unit_test.yml
  93,48:           bash /home/gpadmin/go/src/github.com/greenplum-db/gpbackup/ci/scripts/run_gpbackup_tests.bash

/home/den/gpdb/src/gpbackup/ci/cleanup/plugin-cleanup.yml
  39,29:     uri: https://github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/regression/regression_pipeline.yml
  75,29:     uri: https://github.com/greenplum-db/gpbackup
  83,29:     uri: https://github.com/greenplum-db/gpbackup-s3-plugin
  112,29:     uri: https://github.com/greenplum-db/gpdb
  119,29:     uri: https://github.com/greenplum-db/gpdb

/home/den/gpdb/src/gpbackup/ci/scale/scale_pipeline.yml
  53,29:     uri: https://github.com/greenplum-db/gpbackup
  60,29:     uri: https://github.com/greenplum-db/gpdb

/home/den/gpdb/src/gpbackup/ci/scripts/all-tests.bash
  27,32:   cd \${GOPATH}/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/scripts/build-ddboost.bash
  6,21: mkdir -p /usr/local/greenplum-db-devel
  7,42: tar -xzf bin_gpdb/*.tar.gz -C /usr/local/greenplum-db-devel
  8,19: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/build-go-binaries.bash
  13,32: pushd ${GOPATH}/src/github.com/greenplum-db/gpbackup
  26,32: pushd ${GOPATH}/src/github.com/greenplum-db/gpbackup-s3-plugin

/home/den/gpdb/src/gpbackup/ci/scripts/build-gppkgs.bash
  64,21: mkdir -p /usr/local/greenplum-db-devel
  65,49: tar -xzf bin_gpdb/bin_gpdb.tar.gz -C /usr/local/greenplum-db-devel
  69,19: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/ddboost-plugin-perf.bash
  54,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup/plugins

/home/den/gpdb/src/gpbackup/ci/scripts/ddboost-plugin-tests.bash
  85,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup/plugins
  92,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/scripts/icw-migr-backup.bash
  30,51: mkdir -p \${GOPATH}/bin \${GOPATH}/src/github.com/greenplum-db
  31,49: cp -R $(pwd)/gpbackup \${GOPATH}/src/github.com/greenplum-db/
  34,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup
  38,19: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/icw-migr-restore.bash
  31,51: mkdir -p \${GOPATH}/bin \${GOPATH}/src/github.com/greenplum-db
  32,49: cp -R $(pwd)/gpbackup \${GOPATH}/src/github.com/greenplum-db/
  35,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup
  39,19: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/run_gpbackup_tests.bash
  38,129: PATH=$PATH:/opt/go/bin:~/go/bin GOPATH=~/go make depend build install integration end_to_end -C /home/gpadmin/go/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/scripts/s3-plugin-perf.bash
  41,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup/plugins

/home/den/gpdb/src/gpbackup/ci/scripts/s3-plugin-tests.bash
  71,29:   pushd ~/go/src/github.com/greenplum-db/gpbackup/plugins
  98,29:   pushd ~/go/src/github.com/greenplum-db/gpbackup/plugins

/home/den/gpdb/src/gpbackup/ci/scripts/setup-cluster-env.bash
  7,19: GPHOME=/usr/local/greenplum-db-devel
  11,51:     sudo mkdir -p /home/gpadmin/go/src/github.com/greenplum-db && \
  14,56: scp -r -q gpbackup cdw:/home/gpadmin/go/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/scripts/test-gpbackup-manager.bash
  41,65:     gpssh -f /home/gpadmin/segment_host_list "source /usr/local/greenplum-db-devel/greenplum_path.sh; gpbackup_helper --version"
  47,65:     gpssh -f /home/gpadmin/segment_host_list "source /usr/local/greenplum-db-devel/greenplum_path.sh; gpbackup_helper --version"
  50,37:     pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/scripts/test-on-local-cluster.bash
  41,51: mkdir -p \${GOPATH}/bin \${GOPATH}/src/github.com/greenplum-db
  42,49: cp -R $(pwd)/gpbackup \${GOPATH}/src/github.com/greenplum-db/
  45,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup
  49,19: source /usr/local/greenplum-db-devel/greenplum_path.sh
  82,33: pushd \${GOPATH}/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/tasks/build-go-binaries.yml
  8,27:   path: go/src/github.com/greenplum-db/gpbackup
  10,27:   path: go/src/github.com/greenplum-db/gpbackup-s3-plugin
  22,27:   path: go/src/github.com/greenplum-db/gpbackup/ci/scripts/build-go-binaries.bash

/home/den/gpdb/src/gpbackup/ci/templates/gpbackup-tpl.yml
  283,29:     uri: https://github.com/greenplum-db/gpbackup
  294,29:     uri: https://github.com/greenplum-db/gpbackup-s3-plugin
  325,29:     uri: https://github.com/greenplum-db/gpdb
  335,29:     uri: https://github.com/greenplum-db/gpdb
  345,29:     uri: https://github.com/greenplum-db/gpdb
  698,12:     owner: greenplum-db

/home/den/gpdb/src/gpbackup/ci/Dockerfile
  7,40: COPY . /home/gpadmin/go/src/github.com/greenplum-db/gpbackup

/home/den/gpdb/src/gpbackup/ci/gpbackup-generated.yml
  263,29:     uri: https://github.com/greenplum-db/gpbackup
  271,29:     uri: https://github.com/greenplum-db/gpbackup-s3-plugin
  293,29:     uri: https://github.com/greenplum-db/gpdb
  300,29:     uri: https://github.com/greenplum-db/gpdb
  307,29:     uri: https://github.com/greenplum-db/gpdb

/home/den/gpdb/src/gpbackup/ci/gpbackup-release-generated.yml
  254,29:     uri: https://github.com/greenplum-db/gpbackup
  263,29:     uri: https://github.com/greenplum-db/gpbackup-s3-plugin
  288,29:     uri: https://github.com/greenplum-db/gpdb
  296,29:     uri: https://github.com/greenplum-db/gpdb
  304,29:     uri: https://github.com/greenplum-db/gpdb
  624,12:     owner: greenplum-db
```

</details>

4. Is it ok to use greenplum_path.sh here?

<details>
<summary>spoiler</summary>

```sh
/home/den/gpdb/src/gpbackup/ci/scripts/build-ddboost.bash
  8,38: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/build-gppkgs.bash
  69,38: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/icw-migr-backup.bash
  38,38: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/icw-migr-restore.bash
  39,38: source /usr/local/greenplum-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/run_gpbackup_tests.bash
  29,85: ln -s /usr/local/greengage-db-devel/greengage_path.sh /usr/local/greengage-db-devel/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/setup-cluster-env.bash
  27,22:     source ${GPHOME}/greenplum_path.sh

/home/den/gpdb/src/gpbackup/ci/scripts/test-gpbackup-manager.bash
  41,84:     gpssh -f /home/gpadmin/segment_host_list "source /usr/local/greenplum-db-devel/greenplum_path.sh; gpbackup_helper --version"
  47,84:     gpssh -f /home/gpadmin/segment_host_list "source /usr/local/greenplum-db-devel/greenplum_path.sh; gpbackup_helper --version"

/home/den/gpdb/src/gpbackup/ci/scripts/test-on-local-cluster.bash
  49,38: source /usr/local/greenplum-db-devel/greenplum_path.sh
```

</details>


Should we remove this now?
https://github.com/GreengageDB/gpbackup/blob/45c612b6b077ffd92857c2a64018e945d9180322/ci/scripts/run_gpbackup_tests.bash#L28-L29

whitehawk commented on history/history.go:
```diff
@@ -21,8 +21,8 @@ type RestorePlanEntry struct {
     BackupStatusInProgress = "In Progress"
-	BackupStatusSucceed = "Success"
-	BackupStatusFailed  = "Failure"
+       BackupStatusSucceed = "Success"
```
Looks like not relevant change.

KnightMurloc replied:
fixed

whitehawk commented on SECURITY.md:
```diff
@@ -1,29 +1,29 @@
-(security@greenplum.org).
+If you know of a publicly disclosed security vulnerability for Greengage
+Database, please **IMMEDIATELY** contact the Greengage Database project team
+(security@greengagedb.org).
```
Does this email address exist?

KnightMurloc replied:
We did the same thing when renaming gpdb. Such an address will exist in the future. I think.

whitehawk commented on arenadata/README.md:
```diff
@@ -1,13 +0,0 @@
-## How to run tests
```
The instructions how to run tests are just removed. I think that updated correct instructions should be placed somewhere in ci/README or similar file.

KnightMurloc replied:
> Maybe add link to this info into the main readme.md?

Done.

KnightMurloc commented on .github/workflows/build_and_unit_test.yml:
```diff
@@ -90,4 +90,4 @@ jobs:
           --sysctl 'kernel.sem=500 1024000 200 4096'
           gpbackup:test
-          bash /home/gpadmin/go/src/github.com/greenplum-db/gpbackup/arenadata/run_gpbackup_tests.bash
+          bash /home/gpadmin/go/src/github.com/greenplum-db/gpbackup/ci/scripts/run_gpbackup_tests.bash
```
fixed

KnightMurloc commented:
> Should we change vmware mentions? e.g.:
https://github.com/GreengageDB/gpbackup/blob/45c612b6b077ffd92857c2a64018e945d9180322/CODE-OF-CONDUCT.md#L59

changed

> https://github.com/GreengageDB/gpbackup/blob/45c612b6b077ffd92857c2a64018e945d9180322/CONTRIBUTING.md#L12

> https://github.com/GreengageDB/gpbackup/blob/516109fcb056bf5cf9c97e70c1bc217ee9ffc4f2/ci/regression/README.md#L13

We don't touch ci folder for now.

> What will we do with these test paths? Maybe it should be also changed like ${GOPATH}/src/github.com/GreengageDB/gpbackup?

Same.

> Is it ok to use greenplum_path.sh here?

Same.

> Should we remove this now?

Done.

hilltracer commented:
> From the backup config 9-segment-db-replicated.tar.gz the arenadata suffix has been removed.

Explain please why did you change 9-segment-db-replicated.tar.gz? Why where the other archives not changed? What was the problem related to test?

Is it ok that 9-segment-db-replicated.tar.gz still contains _arenadata?
<details>
<summary>spoiler</summary>

```sh
ADBDEV-8303/9-segment-db-replicated/demoDataDir-1/backups/20231128/20231128092650/gpbackup_20231128092650_config.yaml:
  5  databasename: testdb
  6: databaseversion: 6.25.1_arenadata51+dev.97.g6b28182122 build dev
  7  segmentcount: 9

ADBDEV-8303/9-segment-db-replicated/demoDataDir-1/backups/20231128/20231128092650/gpbackup_20231128092650_report:
  3  timestamp key:         20231128092650
  4: gpdb version:          6.25.1_arenadata51+dev.97.g6b28182122 build dev
  5  gpbackup version:      1.29.4+dev.10.g217a3b8d
```
</details>

KnightMurloc commented:
> Explain please why did you change 9-segment-db-replicated.tar.gz? Why where the other archives not changed? What was the problem related to test?

Only this archive contained the gpdbackup version with the arenadata suffix.

> Is it ok that 9-segment-db-replicated.tar.gz still contains _arenadata?

It shouldn't cause any problems, but I think we can delete it too.

whitehawk commented on ci/README.md:
````diff
@@ -1,10 +1,13 @@
-There are other pipelines, such as those housed in `/regression` and `/scale` that are not generated automatically, and are instead manually maintained.  These still use the same Vault interpolation of secrets and cli branch name conventions when being flown.  Usage of each is described in the relevant directories.
+7x:
+```bash
+docker build -t gpbackup:test7x -f Dockerfile --build-arg GPDB_IMAGE=greengagedb/ggdb7_ubuntu:testing .
````
```suggestion
docker build -t gpbackup:test7x -f Dockerfile --build-arg GGDB_IMAGE=greengagedb/ggdb7_ubuntu:testing .
```
